### DISCUSSION TITLE
Regarding HLS accelerated playback

### DISCUSSION THREAD

**Discussion by kyao777** [bluenviron/mediamtx#2108] [Questions] [answered]
First of all, thank you very much for the kind words about this open-source project. It's a fantastic initiative. However, I'm facing a difficulty now and need everyone's help.

My requirements are as follows:
I'm using mediaMtx to split the RTSP stream from a network camera into hourly TS files, saving them on a Linux server. Now, I want to implement video playback functionality and be able to achieve 2x, 4x, and 8x playback speeds.

Currently, I have successfully implemented the video recording feature. During video playback, I'm using the "runOnDemand" command.
	{
	    "runOnDemand":"ffmpeg -re -ss 120 -i "concat:/home/admins/data/cam68s/s20230406115746.ts|/home/admins/data/cam68s/s20230406125745.ts|/home/admins/data/cam68s/s20230406135745.ts" -c:v copy -f rtsp -rtsp_transport tcp rtsp://localhost:$RTSP_PORT/$RTSP_PATH",
	    "runOnDemandRestart": true
	}
When playing the video at normal (1x) speed in the browser using HLS, there are no issues. But when I try to play it at 2x or 4x speed, the video becomes choppy. I attempted removing the "-re" parameter, but doing so resulted in an error. The log is as follows:

Input #0, mpegts, from 'concat:/home/ceshi/data/cam212s/s20230526172246.ts|/home/ceshi/data/cam212s/s20230526182247.ts':
  Duration: 02:00:01.73, start: 1.400000, bitrate: 6305 kb/s
  Program 1 
    Metadata:
      service_name    : Media Presentation
      service_provider: FFmpeg
  Stream #0:0[0x100]: Video: h264 (Main) ([27][0][0][0] / 0x001B), yuvj420p(pc, bt709, progressive), 2560x1440, 25 fps, 25 tbr, 90k tbn, 50 tbc
  Stream #0:1[0x101]: Audio: mp2 ([4][0][0][0] / 0x0004), 16000 Hz, mono, s16p, 160 kb/s
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (mp2 (native) -> aac (native))
Press [q] to stop, [?] for help
2023/07/25 16:10:34 INF [RTSP] [conn 127.0.0.1:42260] opened
2023/07/25 16:10:35 INF [RTSP] [session 2732148f] created by 127.0.0.1:42260
2023/07/25 16:10:35 INF [RTSP] [session 2732148f] is publishing to path 'pcam1', with TCP, 2 tracks (H264, MPEG4-audio-gen)
Output #0, rtsp, to 'rtsp://localhost:8554/pcam1':
  Metadata:
    encoder         : Lavf58.76.100
  Stream #0:0: Video: h264 (Main) ([27][0][0][0] / 0x001B), yuvj420p(pc, bt709, progressive), 2560x1440, q=2-31, 25 fps, 25 tbr, 90k tbn, 25 tbc
  Stream #0:1: Audio: aac (LC), 16000 Hz, mono, fltp, 69 kb/s
    Metadata:
      encoder         : Lavc58.134.100 aac2023/07/25 16:10:35 INF [HLS] [muxer pcam1] is converting into HLS, 2 tracks (H264, MPEG4-audio-gen)

2023/07/25 16:10:36 INF [HLS] [muxer pcam1] destroyed (**_muxer error: unable to extract DTS: invalid POC_**)
2023/07/25 16:10:36 INF [HLS] [muxer pcam1] created (requested by 192.168.20.191)
2023/07/25 16:10:36 WAR [RTSP] [session 2732148f] invalid FU-A packet (non-starting)
2023/07/25 16:10:36 WAR [RTSP] [session 2732148f] invalid FU-A packet (non-starting)
2023/07/25 16:10:36 WAR [RTSP] [session 2732148f] invalid FU-A packet (non-starting)
2023/07/25 16:10:36 INF [HLS] [muxer pcam1] is converting into HLS, 2 tracks (H264, MPEG4-audio-gen)
2023/07/25 16:10:36 INF [HLS] [muxer pcam1] destroyed (muxer error: unable to extract DTS: invalid POC)
2023/07/25 16:10:37 INF [HLS] [muxer pcam1] created (requested by 192.168.20.191)
2023/07/25 16:10:37 INF [HLS] [muxer pcam1] is converting into HLS, 2 tracks (H264, MPEG4-audio-gen)
**_frame=27535 fps=5062 q=-1.0 Lsize=N/A time=00:18:21.36 bitrate=N/A speed= 202x_**    
video:800980kB audio:9291kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
[aac @ 0x55a86fafa740] Qavg: 124.545
2023/07/25 16:10:40 INF [path pcam1] runOnDemand command stopped
2023/07/25 16:10:40 INF [RTSP] [session 2732148f] destroyed (torn down by 127.0.0.1:42260)
2023/07/25 16:10:40 INF [RTSP] [conn 127.0.0.1:42260] closed (EOF)
2023/07/25 16:10:40 INF [HLS] [muxer pcam1] destroyed (terminated)
2023/07/25 16:10:40 INF [HLS] [muxer pcam1] created (requested by 192.168.20.191)
2023/07/25 16:10:40 INF [path pcam1] runOnDemand command started

I hope someone can help me solve this problem. Thank you very much.

**Comment by aler9** [comment, marked as answer]
Hello, if you send a video stream to the server at twice the original speed, the bandwidth of the stream will be twice the original, and so on, until all the available bandwidth of your server is occupied. RTSP is not fit for playback since it's a "real time streaming protocol": when some data is lost due to high bandwidth consumption, the server tries to recover it, but after a limited number of retries, it discards it in order not to delay the stream (since the stream must be real-time).

The same applies to SRT and WebRTC.

My advice is to save in MP4 format and serve the files with an HTTP server. All modern browsers are able to read MP4 files, set the playback speed on the client side, pause the video in case the download speed is too slow and resume the video when the next portion of the video is available (that's exactly how YouTube works).

**Reply by kyao777** [reply]
Thank you very much for your reply. I have already followed the instructions and done it that way.

[a comment by an automated account is not shown]
